pkg/flare/archive_win.go

```diff
@@ -209,3 +210,52 @@ func (p permissionsInfos) add(filePath string) {}
 func (p permissionsInfos) commit(tempDir, hostname string, mode os.FileMode) error {
 	return nil
 }
+
+func zipServiceStatus(tempDir, hostname string) error {
+	f := filepath.Join(tempDir, hostname, "servicestatus.txt")
+	err := ensureParentDirsExist(f)
+	if err != nil {
+		log.Warnf("Error in ensureParentDirsExist %v", err)
+		return err
+	}
+
+	fh, err := os.Create(f)
+	if err != nil {
+		log.Warnf("Error creating temp file %s %v", f, err)
+	}
+	defer fh.Close()
+	cancelctx, cancelfunc := context.WithTimeout(context.Background(), execTimeout)
+	defer cancelfunc()
+
+	cmd := exec.CommandContext(cancelctx, "powershell", "-c", "get-service", "data*,ddnpm", "|", "fl")
+
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err = cmd.Run()
+	if err != nil {
+		log.Warnf("Error running powershell command %v", err)
+		// for some reason the lodctr command returns error 259 even when
+		// it succeeds.  Log the error in case it's some other error,
+		// but continue on.
+	}
+
+	_, err = fh.Write(out.Bytes())
+	if err != nil {
+		log.Warnf("Error writing file %v", err)
+		return err
+	}
+	// compute the location of the driver
+	ddroot, err := winutil.GetProgramFilesDirForProduct("DataDog Agent")
+	if err == nil {
+		pathtodriver := filepath.Join(ddroot, "bin", "agent", "driver", "ddnpm.sys")
+		fi, err := os.Stat(pathtodriver)
+		if err != nil {
+			fh.WriteString(fmt.Sprintf("Failed to stat file %v %v\n", pathtodriver, err))
```

Comment:
Why is this writing to the file, rather than returning the error?

Reply:
because it's important information that I want in the flare; the fact that the driver file isn't present.